### ISSUE TITLE
Sass variable in CSS variable outputted literally

### ISSUE THREAD

**Issue by thany** [sass/libsass#2835]
[todo]: # (Title: Be as meaningful as possible)
[todo]: # (Title: Try to use 60 or less chars)

[todo]: # (This is only a template!)
[todo]: # (remove unneeded bits)
[todo]: # (use github preview!)

## input.scss

[todo]: # (always test and report with scss syntax)
[todo]: # (use sass only when results differ from scss)

```scss
$var: 4px;

div {
    --var: $var;
}
```

## Actual results

[todo]: # (update version info!)

LibSass 3.5.2/3.5.4 [1] [2]
```css
div {
  --var: $var;
}
```

## Expected result

[todo]: # (update version info!)

ruby sass 3.4.25, 3.5.6
```css
div {
  --var: 4px;
}
```

[todo]: # (update version info!)
[todo]: # (example for node-sass!)

version info:
LibSass 3.5.2 on Sassmeister

[todo]: # (Go to http://libsass.ocbnet.ch/srcmap)
[todo]: # (Enter your SCSS code and hit compile)
[todo]: # (Click `bookmark` and replace the url)
[todo]: # (link is used in actual results above)

[1]: http://libsass.ocbnet.ch/srcmap/#JHZhcjogNHB4OwoKZGl2IHsKICAgIC0tdmFyOiAkdmFyOwp9
[2]: https://www.sassmeister.com/gist/f67f4d868ac7bbc1402b8dea8e6dd4f4

**Comment by glebm** [comment]
Works as intended, see https://github.com/sass/sass/issues/2383#issuecomment-336349172

**Comment by thany** [comment]
If it really works as intended, then the intention is wrong 😃

Again, **it works fine in Ruby Sass**, and it also works fine for any regular property. It also works fine for properties that start with a single dash. Since CSS variables are gaining traction, they oughta be supported. It currently feels like CSS variables are somehow magically exempted from the Sass parser.

Question: what is the reasoning behind this behaviour?

**Comment by xzyfer** [comment]
See https://github.com/sass/sass/issues/1128

**Comment by thany** [comment]
That appears to be an issue about using native CSS variables syntax as if they were Sass variables. That would be a feature request that's very hard to implement.

What I'm filing here, is an actual *bug*.

**Comment by xzyfer** [comment]
It's not a bug. As the link issue explains the CSS spec is worded such that any value is a valid value for a custom property. This includes valid Sass values like Sass variables.

As such the Sass team made the choice to treat custom property values as plain CSS strings except for interpolations `#{}`. So if you want to assign a Sass variable to a custom property it must be inside an interpolation.

**Comment by thany** [comment]
So the "bug" is in Ruby Sass?

**Comment by xzyfer** [comment]
Possibly in those older versions of Ruby Sass. I suggest trying the dart sass implementation at sassmeister.com as it is now the canonical.

**Comment by thany** [comment]
Alright.

So will this ever be fixed?

As much as it it intended behaviour, it is not desired behaviour.
I can't really think of a reason why having `--var: $var` in your output css would be helpful in any way... I also still don't understand why Sass makes an arbitrary decision to ignore everything in a property that starts with `--`

**Comment by glebm** [comment]
> I can't really think of a reason why having

This decision is not arbitrary. I suggested you actually read the issue comment and discussion that I've linked to above (https://github.com/sass/sass/issues/2383#issuecomment-336349172).

If you'd like to discuss the *spec* itself, the correct repo for that is https://github.com/sass/sass-spec.

This repo is just for the libsass implementation. It's not even the "reference" implementation (that's dart-sass) and none of the maintainers here work on the Sass specification.
